include/nuttx/sched.h

```diff
@@ -530,14 +530,6 @@ struct task_group_s
   FAR struct join_s *tg_jointail; /* Tail of a list of join data            */
 #endif
 
-  /* Thread local storage ***************************************************/
-
-#if CONFIG_TLS_NELEM > 0
-  tls_ndxset_t tg_tlsset;                   /* Set of TLS indexes allocated */
-
```

Comment:
I think that tg_tlsset should remain inside the group structure.  It is not accessible to user's and should be protected.  Access must be atomic and this was done with a critical section, I believe.

Reply:
Nevermind... I see you replaced with critical section with a semaphore.